INSTALL.md

```diff
@@ -87,3 +87,94 @@ environment using:
 
     conda deactivate
     conda env remove -n compiler_gym
+
+## Building from source with CMake
+
+### Dependency instructions for Ubuntu
+
+```bash
+sudo apt-get install clang-9 clang++-9 lld-9 patchelf tar bzip2 \
+  autoconf libtool make ninja-build ccache \
+  python3 python3-pip python3-venv \
+  apt-transport-https curl gnupg wget \
+```
+
+Requires CMake (==3.20).
+
+```bash
+wget https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.sh -O cmake.sh
+bash cmake.sh --prefix=$HOME/.local --exclude-subdir --skip-license
+rm cmake.sh
+```
+
+Setup a virtual environment.
+
+```bash
+python3 -m venv compiler_gym_venv
+source compiler_gym_venv/bin/activate
+pip install --upgrade pip
+# in the source directory
+python -m pip install -r requirements.txt
+```
+
+To build programl internally it requires Bazel.
+
+```
+curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
+sudo apt update && sudo apt install bazel-4.0.0
+```
```

Comment:
This could be consolidated with the other "building from source" instructions above which cover setting up a conda environment and installing bazel.

(this doesn't need fixing right now, just a note to future me :) )

Reply:
Done.